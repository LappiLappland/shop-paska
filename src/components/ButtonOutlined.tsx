import { ReactNode, useRef } from 'react';
import useRipple from '../hooks/useRipple';

interface ButtonProps {
  children?: ReactNode;
  className?: string;
  rippleClassName?: string;
  addPadding?: boolean;
  type?: 'submit' | 'reset' | 'button';
  disabled?: boolean;
  dataTestid?: string;
  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

export default function ButtonOutlined({
  children,
  className = '',
  rippleClassName = '',
  addPadding = true,
  type,
  disabled,
  dataTestid,
  onClick,
}: ButtonProps) {
  const buttonRef = useRef<HTMLButtonElement>(null);

  const ripple = useRipple(buttonRef, {
    color: 'bg-primary/focused',
    className: rippleClassName,
  });

  return (
    <button
      ref={buttonRef}
      data-testid={dataTestid}
      className={`
      h-10 ${addPadding ? 'px-3' : ''} hover:bg-border-primary/hovered disabled:text-on-surface-r relative
      flex items-center
      justify-center
      border border-outline bg-transparent fill-primary stroke-primary
      text-primary transition-colors
      hover:bg-primary/hovered-r focus-visible:border-primary/focused
      focus-visible:bg-primary/focused-r
      active:bg-primary/pressed-r disabled:border-on-surface/disabled-r disabled:fill-on-surface/disabled-r disabled:stroke-on-surface/disabled-r
      ${className}`}
      type={type}
      disabled={disabled}
      onClick={(e) => {
        if (onClick) onClick(e);
      }}
    >
      {children}
      {ripple}
    </button>
  );
}
