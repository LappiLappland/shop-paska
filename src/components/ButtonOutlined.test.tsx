import { act, render } from '@testing-library/react';
import ButtonOutlined from './ButtonOutlined';

describe('ButtonOutlined component', () => {
  test('renders button with default state', () => {
    const { container } = render(<ButtonOutlined>Click me</ButtonOutlined>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders button with disabled state', () => {
    const { container } = render(
      <ButtonOutlined disabled>Click me</ButtonOutlined>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders button with padding disabled', () => {
    const { container } = render(
      <ButtonOutlined addPadding={false}>Click me</ButtonOutlined>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('calls onClick handler when button is clicked', () => {
    const onClick = jest.fn();
    const { getByRole } = render(
      <ButtonOutlined onClick={onClick}>Click me</ButtonOutlined>
    );
    const button = getByRole('button');
    act(() => {
      button.click();
    })
    expect(onClick).toHaveBeenCalled();
  });
});
