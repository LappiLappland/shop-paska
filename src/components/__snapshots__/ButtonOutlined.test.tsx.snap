// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonOutlined component renders button with default state 1`] = `
<button
  class="
      h-10 px-3 hover:bg-border-primary/hovered disabled:text-on-surface-r relative
      flex items-center
      justify-center
      border border-outline bg-transparent fill-primary stroke-primary
      text-primary transition-colors
      hover:bg-primary/hovered-r focus-visible:border-primary/focused
      focus-visible:bg-primary/focused-r
      active:bg-primary/pressed-r disabled:border-on-surface/disabled-r disabled:fill-on-surface/disabled-r disabled:stroke-on-surface/disabled-r
      "
>
  Click me
  <span
    class="absolute inset-0 h-full w-full overflow-hidden "
  />
</button>
`;

exports[`ButtonOutlined component renders button with disabled state 1`] = `
<button
  class="
      h-10 px-3 hover:bg-border-primary/hovered disabled:text-on-surface-r relative
      flex items-center
      justify-center
      border border-outline bg-transparent fill-primary stroke-primary
      text-primary transition-colors
      hover:bg-primary/hovered-r focus-visible:border-primary/focused
      focus-visible:bg-primary/focused-r
      active:bg-primary/pressed-r disabled:border-on-surface/disabled-r disabled:fill-on-surface/disabled-r disabled:stroke-on-surface/disabled-r
      "
  disabled=""
>
  Click me
  <span
    class="absolute inset-0 h-full w-full overflow-hidden "
  />
</button>
`;

exports[`ButtonOutlined component renders button with padding disabled 1`] = `
<button
  class="
      h-10  hover:bg-border-primary/hovered disabled:text-on-surface-r relative
      flex items-center
      justify-center
      border border-outline bg-transparent fill-primary stroke-primary
      text-primary transition-colors
      hover:bg-primary/hovered-r focus-visible:border-primary/focused
      focus-visible:bg-primary/focused-r
      active:bg-primary/pressed-r disabled:border-on-surface/disabled-r disabled:fill-on-surface/disabled-r disabled:stroke-on-surface/disabled-r
      "
>
  Click me
  <span
    class="absolute inset-0 h-full w-full overflow-hidden "
  />
</button>
`;
